Move demangle filter to an optional module

bamx23 opened the pull request:
Demangling is not a requirement for crash reports that are later passed through the symbolicator (with dSYMs).
To reduce minimal library size and exclude parts of swift/llvm code base, this module becomes optional.

GLinnik21 commented on Package.swift:
```diff
@@ -217,6 +213,21 @@ let package = Package(
         ]
     ),
 
+    .target(
```
I think we need PrivacyReport in every target

GLinnik21 commented on Sources/KSCrashDemangleFilter/KSCrashReportFilterDemangle.m:
```diff
@@ -0,0 +1,168 @@
+    return [result copy];
+}
+
+- (void)filterReports:(NSArray<id<KSCrashReport>> *)reports onCompletion:(KSCrashReportFilterCompletion)onCompletion
```
I think marking which code refers to the `KSCrashReportFilter` protocol would make the code easier to understand.

GLinnik21 reviewed (approved):
Great! Now we will be more in line with the principle of single responsibility!

GLinnik21 commented:
I'm also curious about how users will install KSCrash by default, as they might have expected that demangling worked by default. It seems like most of the users would still need it, so I'm wondering how we can make that as easy as possible for them.

bamx23 commented:
> I'm also curious about how users will install KSCrash by default, as they might have expected that demangling worked by default. It seems like most of the users would still need it, so I'm wondering how we can make that as easy as possible for them. Also, I think we should update README as well.

It's a valid concern. I've added demangling to Installations API and made it enabled by-default. So those who were using installations can keep this functionality without extra efforts. And I've updated the README to highlight that this is now not a default behavior.

Also, we'll need a "migration guide" for 2.0 anyway as there are other breaking changes and we need to highlight the expected app code updates.
